site/en/tutorials/sequences/text_generation.ipynb

```diff
@@ -861,7 +861,7 @@
     "\n",
     "    # using a multinomial distribution to predict the word returned by the model\n",
     "    predictions = predictions / temperature\n",
-    "    predicted_id = tf.multinomial(predictions, num_samples=1)[0][0].numpy()\n",
+    "    predicted_id = tf.multinomial(predictions[:-1], num_samples=1)[0][0].numpy()\n",
```

Comment:
Shouldn't it be 
tf.multinomial(predictions, num_samples=1)[-1][0].numpy() ?

Reply:
This is a good point... something is funny right here.

The model is violating the [batch, sequence, depth] shape convention.

I'll fix it give me a minute.